### PR TITLE
feat(kernel): wire audit chain verification into kernel boot

### PR DESCRIPTION
**Pull request by joshuajbouw**
## Summary

- Open persistent `AuditLog` at `~/.astrid/audit.db` during `Kernel::new()` with runtime ed25519 signing key at `~/.astrid/keys/runtime.key` (generated on first boot, `0o600` permissions)
- Run `verify_all()` on startup to detect historical chain tampering - logs `error!` per issue but does not block boot (fail-open for availability, loud alert for integrity)
- Add 7 negative/edge-case tests for chain verification: tampered signature, broken link, invalid genesis, multi-session tamper detection, empty log, key rotation forward-verifiability
- Add 4 unit tests for `load_or_generate_runtime_key`: creation, idempotency, bad length rejection, Unix permissions

## Test Plan

- `cargo test -p astrid-audit` - 16 tests (7 new negative/edge-case tests)
- `cargo test -p astrid-kernel` - 4 tests (all new, covering key management)
- `cargo test --workspace` - full green, no regressions
- `cargo clippy --workspace -- -D warnings` - clean

## Risk Traceability

| Risk | Test | Verified? |
|---|---|---|
| Tampered signature detection | `test_verify_detects_tampered_signature` | Yes |
| Broken chain link detection | `test_verify_detects_broken_link` | Yes |
| Invalid genesis detection | `test_verify_detects_invalid_genesis` | Yes |
| Multi-session tamper isolation | `test_verify_all_detects_tampered_session` | Yes |
| Empty audit log on first boot | `test_verify_empty_log_is_valid` | Yes |
| Key file wrong length | `test_load_or_generate_rejects_bad_key_length` | Yes |
| Key generation idempotency | `test_load_or_generate_is_idempotent` | Yes |
| Key file permissions (Unix) | `test_load_or_generate_sets_secure_permissions` | Yes |
| Key rotation forward-verifiability | `test_key_rotation_entries_verify_via_embedded_pubkey` | Yes |

## Related Issues

- #305 (OOS: `block_on` thread spawning in SurrealKV storage - pre-existing, not introduced here)

### CODE REVIEW

File: crates/astrid-audit/src/log.rs [hunk 1]
```diff
@@ -504,4 +504,263 @@ mod tests {
...
+                        args_hash: ContentHash::zero(),
+                    },
+                    AuthorizationProof::NotRequired {
+                        reason: "test".to_string(),
+                    },
+                    AuditOutcome::success(),
+                )
+                .unwrap();
+            ids.push(id);
+        }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-audit/src/log.rs]: @gemini-code-assist Good catch. Extracted `append_test_entries()` helper and replaced all 5 duplicated chain-building loops. Pushed.

File: crates/astrid-kernel/src/lib.rs [hunk 2]
```diff
@@ -250,6 +258,93 @@ impl Kernel {
...
+                    total_sessions,
+                    tampered_sessions = tampered,
+                    "Audit chain verification found tampered sessions"
+                );
+            } else if total_sessions > 0 {
+                tracing::info!(
+                    total_sessions,
+                    "Audit chain verification passed for all sessions"
+                );
+            }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-kernel/src/lib.rs:310]: @gemini-code-assist Agreed - consolidated into a single pass with a `tampered_sessions` counter incremented inside the issue-logging loop. Pushed.
